### PR TITLE
Updated Arrays reference page.

### PR DESCRIPTION
**Pull request by tomball**
Here's a draft of the Arrays doc. Please review and let me know when it's suitable to pull.

### CODE REVIEW

File: docs/Arrays.md [hunk 1]
```diff
@@ -3,6 +3,67 @@ title: Arrays
...
+direct access methods for accessing and replacing regions of its elements buffer.
+
+In addition to access methods, IOSPrimitive.h defines access functions that avoid message passing
+overhead. "Get" returns the element at a specified index, "GetRange" returns a specified range of
+elements, and "SetRange" replaces a specified range of elements. Each function starts with the type
+name, followed by the function's action; for example, `IOSByteArray_Get(array, index)`.
+
+The "GetRef" functions are similar to "Get" functions, but return a pointer to the specified
+element. This allows faster access to read or modify the contents of an array using C expressions.
+Since this access has no range, bounds, or type checking, *it is critical that these functions be
```
Comment (kstanger) [inline docs/Arrays.md]: "Get" and "GetRef" do check bounds.

File: docs/Arrays.md [hunk 2]
```diff
@@ -3,6 +3,67 @@ title: Arrays
...
+
+### Object Arrays
+
+Object arrays are similar to primitive arrays, with a single
+[IOSObjectArray](https://github.com/google/j2objc/blob/master/jre_emul/Classes/IOSObjectArray.h)
+representing all non-primitive arrays. `IOSObjectArray` retains and releases its elements, like
+`NSMutableArray` does.
+
+In addition to the functions defined for primitive arrays, `IOSObjectArray` adds a "SetAndConsume"
+function. This function releases the object being set if an exception is thrown by the function.
```
Comment (kstanger) [inline docs/Arrays.md]: This is confusing. I'd say something like: "This function provides consume semantics (ns_consumed) where the caller must provide a retained item." You can link to the ns_consumed documentation here: http://clang-analyzer.llvm.org/annotations.html#attr_ns_consumed

File: docs/Arrays.md (hunk 1 above)
Comment (tomball) [inline docs/Arrays.md]: Fixed.

File: docs/Arrays.md (hunk 2 above)
Comment (tomball) [inline docs/Arrays.md]: Updated.

**Pull request merged**